### ISSUE TITLE
AdamWOptimizer and learning rate decay

### ISSUE THREAD

**Issue by benleetownsend** [tensorflow/tensorflow#22763]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Linux Ubuntu 16.04 
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**1.11:
- **Python version**:3.6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
I think I have found that the contrib.opt.AdamWOptimizer and associated decoupled weight decay optimizers do not function correctly when using learning rate decay without explicitly applying the decay to the weight_decay parameter. 

In the original paper, as seen in algorithm 2, the schedule multiplier is factored out and applied to the whole expression, the current interface means you have to do `AdamWOptimizer(weight_decay=wd*decay, learning_rate=lr*decay)` to achieve parity with the paper, this is not in its self an issue, but I think the documentation should reflect this difference. Alternatively the API could give a schedule multiplier parameter and then fixed lr and wd parameters used. 

Happy to submit a PR if someone can advise whether a documentation or interface update is the desired approach.

**Comment by facaiy** [comment]
@PhilJd @alextp Could you please comment?

**Comment by JoshuaRM** [comment]
Hello, I am a student at Seneca College, participating in Hacktoberfest. I've read through this issue, and I can improve the documentation to reflect the necessary changes, and can send the pull request. Please let me know if this would be helpful, thanks!

**Comment by facaiy** [comment]
@JoshuaRM Contribution is always welcome. Please link your PR to the issue :-)

**Comment by Ouwen** [comment]
Ah, I was wondering why when my learning rate decreases all my weights went to 0.

**Comment by Ouwen** [comment]
>   Note: when applying a decay to the learning rate, be sure to manually apply
>   the decay to the `weight_decay` as well. For example:
>    ```python
>     decay = tf.train.piecewise_constant(tf.train.get_global_step(), 
>                                         [10000, 15000], [1e-1, 1e-2, 1e-3])
>     lr = 1*decay
>     wd = 1e-4*decay
>      # ...
>      optimizer = tf.contrib.opt.MomentumWOptimizer(learning_rate=lr,
>                                                   weight_decay=wd,
>                                                   momentum=0.9,
>                                                   use_nesterov=True)
>   ```

I can add this snippet to the other weight decay optimizers as well if this looks good.

**Comment by alextp** [comment]
Note that tf.train.piecewise_constant will in tf 2.0 return a function which, when called, returns a tensor (to make things a little safer wrt to multiple calls to optimizer.minimize in a single session.run and eager execution) so you'd have to do something like `wd = lambda: 1e-4 * decay()`.

**Comment by erikchwang** [comment]
When using warm up, do we need to warm up weight_decay together with learning_rate?

**Comment by PhilJd** [comment]
I guess it's very reasonable to do so. If the weight decay is too large compared to the optimizer update (which would be the case for learning warm up without warmed up decay) your weights are likely to become zero.

**Comment by acocac** [comment]
It's still confusing for me how should I declare decay in AdamWOptimizer? I'm using TF 1.14 because is the most recent runtime in Google AI which is the platform to train my models. 

In the repo below a schedule is created according to the number of iterations and batchsize, https://github.com/PhilJd/AdamW_benchmark/blob/4e9bc846f3b913f727f7585389b59fe46aed0056/model/resnet.py

I would appreciate any advice how to set **first_decay_steps** in particular in a hyperparameter optimization task in which the batch size is part of the search grid.

**Comment by mohantym** [comment]
Hi @benleetownsend !
It seems you are using older versions(1.x versions) of Tensorflow which is not supported any more.  Attaching relevant threads for Reference. Ref [1](https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/AdamW), [2](https://www.tensorflow.org/api_docs/python/tf/compat/v1/train/exponential_decay) . Thank you!
